### PR TITLE
Fix load signed integer

### PR DESCRIPTION
**Pull request by yzhang90**

### CODE REVIEW

File: data.md [hunk 1]
```diff
@@ -545,50 +545,52 @@ The strings to connect needs to be unescaped before concatenated, because the `u
...
-    rule                  #get(BM, KEY) => 0        requires notBool KEY inBytes BM
+    syntax Bytes ::= #getBytesRange ( Bytes , Int , Int ) [function, functional]
+ // ----------------------------------------------------------------------------
+    rule #getBytesRange(_, START, WIDTH) => .Bytes
+      requires notBool (WIDTH >=Int 0 andBool START >=Int 0)
+
+    rule #getBytesRange(BM, START, WIDTH) => substrBytes(padRightBytes(BM, START +Int WIDTH, 0), START, START +Int WIDTH)
+      requires START >=Int 0 andBool WIDTH >=Int 0 andBool START <Int lengthBytes(BM)
+
+    rule #getBytesRange(_, _, WIDTH) => padRightBytes(.Bytes, WIDTH, 0) [owise]
```
Comment (hjorthjort) [inline data.md:585]: Why is this one needed, can't you just use the first rule for this case as well?

Comment (hjorthjort) [reply data.md:585]: Well, I guess it makes sense from a semantic viewpoint. But I'd prefer if we used an explicit condition, like
`requires 0 <=Int START andBool 0 <=Int WIDTH andBool  lengthBytes(BM) <=Int START`

File: data.md [hunk 2]
```diff
@@ -545,50 +545,53 @@ The strings to connect needs to be unescaped before concatenated, because the `u
     rule #DS2Bytes(DS) => String2Bytes(#concatDS(DS))
 ```
 
-### Byte Map
+### Local Memory
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply data.md]: done

File: data.md [hunk 3]
```diff
@@ -545,50 +545,53 @@ The strings to connect needs to be unescaped before concatenated, because the `u
...
 ```k
-    syntax Bytes ::= #setRange(Bytes, Int, Int, Int) [function, functional, smtlib(setRange)]
- // -----------------------------------------------------------------------------------------
-    rule                       #setRange(BM, ADDR, VAL, WIDTH) => BM                                                                                 requires notBool (0 <Int WIDTH andBool 0 <=Int VAL andBool 0 <=Int ADDR)
-    rule [setRange-Positive] : #setRange(BM, ADDR, VAL, WIDTH) => #setRange(#set(BM, ADDR, VAL modInt 256), ADDR +Int 1, VAL /Int 256, WIDTH -Int 1) requires          0 <Int WIDTH andBool 0 <=Int VAL andBool 0 <=Int ADDR
-```
+    syntax Bytes ::= #setBytesRange ( Bytes , Int , Bytes ) [function, functional]
+ // ------------------------------------------------------------------------------
+    rule #setBytesRange(BM, START, BS) => replaceAtBytes(padRightBytes(BM, START +Int lengthBytes(BS), 0), START, BS)
+      requires START >=Int 0
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply data.md]: done

File: data.md [hunk 4]
```diff
@@ -545,50 +545,53 @@ The strings to connect needs to be unescaped before concatenated, because the `u
...
- // --------------------------------------------------------------------------------
-    rule [get-Existing] : #get(BM, KEY) => BM [KEY] requires         KEY inBytes BM
-    rule                  #get(BM, KEY) => 0        requires notBool KEY inBytes BM
+    syntax Bytes ::= #getBytesRange ( Bytes , Int , Int ) [function, functional]
+ // ----------------------------------------------------------------------------
+    rule #getBytesRange(_, START, WIDTH) => .Bytes
+      requires notBool (WIDTH >=Int 0 andBool START >=Int 0)
+
+    rule #getBytesRange(BM, START, WIDTH) => substrBytes(padRightBytes(BM, START +Int WIDTH, 0), START, START +Int WIDTH)
+      requires START >=Int 0 andBool WIDTH >=Int 0 andBool START <Int lengthBytes(BM)
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply data.md]: done

File: data.md [hunk 5]
```diff
@@ -545,50 +545,53 @@ The strings to connect needs to be unescaped before concatenated, because the `u
...
+
+    rule #getBytesRange(BM, START, WIDTH) => padRightBytes(.Bytes, WIDTH, 0)
+      requires START >=Int 0 andBool WIDTH >=Int 0 andBool lengthBytes(BM) <=Int START
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply data.md]: done

File: data.md (hunk 1 above)
Comment (yzhang90) [reply data.md]: done

**Review by hjorthjort — APPROVED** [review]

**Pull request merged**